feed/feed.go

```diff
@@ -137,11 +126,80 @@ func chronologicalGenerator(opts chronologicalGeneratorOpts) GenerateFunc {
 			CursorTime: cursorTime,
 		}
 
-		posts, err := pgxStore.ListPostsForNewFeed(ctx, params)
+		storePosts, err := pgxStore.ListPostsForNewFeed(ctx, params)
 		if err != nil {
 			return nil, fmt.Errorf("executing ListPostsForNewFeed: %w", err)
 		}
-		return PostsFromStorePosts(posts), nil
+
+		posts := make([]Post, 0, len(storePosts))
+		for _, p := range storePosts {
+			posts = append(posts, Post{
+				URI:    p.URI,
+				Cursor: bluesky.FormatTime(p.IndexedAt.Time),
+			})
+		}
+
+		return posts, nil
+	}
+}
+
+func hotnessGenerator(alg string, opts generatorOpts) GenerateFunc {
+	return func(ctx context.Context, pgxStore *store.PGXStore, cursor string, limit int) ([]Post, error) {
+		type hotnessCursor struct {
```

Comment:
I'm a tad worried about poorly implemented clients not handling a JSON cursor well (e.g not escaping stuff properly) - maybe we need to base64 url encode this before we return it.